test/OpenTelemetry.Instrumentation.AWSLambda.Tests/Implementation/AWSLambdaHttpUtilTests.cs

```diff
@@ -0,0 +1,170 @@
+
+            var expectedTags = new Dictionary<string, object>
+            {
+                { "http.scheme", "https" },
+                { "http.target", "/path/test" },
+                { "net.host.name", "localhost" },
+                { "net.host.port", 1234 },
+                { "http.method", "GET" },
+            };
+
+            AssertTags(expectedTags, actualTags);
+        }
+
+        [Fact]
+        public void SetHttpTagsFromResult_APIGatewayProxyResponse_SetsCorrectTags()
+        {
+            var response = new APIGatewayProxyResponse
+            {
+                StatusCode = 200,
+            };
+            var activityProcessor = new Mock<BaseProcessor<Activity>>();
+
+            using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddProcessor(activityProcessor.Object)
+                .AddSource("TestActivitySource")
+                .Build();
+
+            using var testActivitySource = new ActivitySource("TestActivitySource");
+            using var activity = testActivitySource.StartActivity("TestActivity");
+
+            AWSLambdaHttpUtil.SetHttpTagsFromResult(activity, response);
+
+            var expectedTags = new Dictionary<string, string>
+            {
+                { "http.status_code", "200" },
+            };
+            AssertTags(expectedTags, activity.Tags);
+        }
+
+        [Fact]
+        public void SetHttpTagsFromResult_APIGatewayHttpApiV2ProxyResponse_SetsCorrectTags()
+        {
+            var response = new APIGatewayHttpApiV2ProxyResponse
+            {
+                StatusCode = 200,
+            };
+            var activityProcessor = new Mock<BaseProcessor<Activity>>();
+
+            using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+                .AddProcessor(activityProcessor.Object)
+                .AddSource("TestActivitySource")
+                .Build();
+
+            using var testActivitySource = new ActivitySource("TestActivitySource");
+            using var activity = testActivitySource.StartActivity("TestActivity");
+
+            AWSLambdaHttpUtil.SetHttpTagsFromResult(activity, response);
+
+            var expectedTags = new Dictionary<string, string>
+            {
+                { "http.status_code", "200" },
+            };
+            AssertTags(expectedTags, activity.Tags);
+        }
+
+        [Theory]
+        [InlineData(null, null, null, null)]
+        [InlineData("", "", "", null)]
+        [InlineData(null, "localhost:4321", "localhost", 4321)]
+        [InlineData(null, "localhost:4321, myhost.com:9876", "localhost", 4321)]
+        [InlineData(null, "localhost", "localhost", null)]
+        [InlineData("http", "localhost", "localhost", 80)]
+        [InlineData("https", "localhost", "localhost", 443)]
+        public void GetHostAndPort_HostHeader_ReturnsCorrectHostAndPort(string httpSchema, string hostHeader, string expectedHost, int? expectedPort)
```

Comment:
Please also test the handling of unparseable (non-integer) ports.

Reply:
Add non-integer ports case.